contrib/enfuseAdvanced.lua

```diff
@@ -612,434 +210,851 @@ local function GetFileName(full_path)
+		dt.preferences.write(mod, 'active_current_preset_ind', 'integer', self.selected)
+	end	
+}
+GUI.Presets.load = dt.new_widget("button"){
+	label = 'laod fusion preset',
+	tooltip ='load current fusion parameters from selected preset',
+	clicked_callback = function() LoadFromPreference(GUI.Presets.current_preset.value) end
+}
+GUI.Presets.save = dt.new_widget("button"){
+	label = 'save to fusion preset',
+	tooltip ='save current fusion parameters to selected preset',
+	clicked_callback = function() SaveToPreference(GUI.Presets.current_preset.value) end
+}
+GUI.Presets.variants = dt.new_widget('check_button'){
+	label = 'create image variants from presets', 
+	value = dt.preferences.read(mod, 'active_variants', 'bool'),
+	tooltip ='create multiple image variants based on the three different presets of the specified type',	
+	clicked_callback = function(self) 
+		dt.preferences.write(mod, 'active_variants', 'bool', self.value)
+		if self.value then
+			GUI.Target.on_conflict.selected = 1
+			GUI.Target.on_conflict.sensitive = false
+			GUI.Presets.variants_type.sensitive = true
+		else
+			GUI.Target.on_conflict.sensitive = true
+			GUI.Presets.variants_type.sensitive = false
+		end
+	end,
+	reset_callback = function(self) self.value = false end
+}
+temp = dt.preferences.read(mod, 'active_variants_type_ind', 'integer')
+if not InRange(temp, 1, 2) then temp = 1 end
+GUI.Presets.variants_type = dt.new_widget('combobox'){
+	label = "create variants type",
+	tooltip = "preset type to be used when creating image variants",
+	selected = temp,
+	'dri', 'dff',
+	changed_callback = function(self)
+		dt.preferences.write(mod, 'active_variants_type', 'string', self.value) 
+		dt.preferences.write(mod, 'active_variants_type_ind', 'integer', self.selected)
+	end,
+	reset_callback = function(self) 
+		self.selected = 1
+		dt.preferences.write(mod, 'active_variants_type', 'string', self.value) 
+		dt.preferences.write(mod, 'active_variants_type_ind', 'integer', self.selected)
+	end	
+}
+GUI.Presets.variants_type.sensitive = GUI.Presets.variants.value
+temp = GUI.Target.format.value
+if temp == 'tif' then temp = 1
+elseif temp == 'jpg' then temp = 2
+else temp = 3
+end
+stack_compression.active = temp
+for _,widget in pairs(GUI.AIS) do
+	widget.sensitive = GUI.align.value
 end
-local enfuse_path_widget = dt.new_widget("file_chooser_button"){
-	title = "Select enfuse[.exe] file",
-	value = enfuse_path,
-	is_directory = false,
+GUI.Target.apply_style.sensitive = GUI.Target.auto_import.value
+GUI.Target.copy_tags.sensitive = GUI.Target.auto_import.value
+GUI.Target.add_tags.sensitive = GUI.Target.auto_import.value
+
+local box_AIS = dt.new_widget('box'){
+	orientation = 'vertical',
+    label_AIS_options,
+	GUI.align,
+	GUI.AIS.radial_distortion,
+	GUI.AIS.optimize_field,
+	GUI.AIS.optimize_image_center,
+	GUI.AIS.auto_crop,
+	GUI.AIS.distortion,
+	GUI.AIS.gpu,
+	GUI.AIS.grid_size,
+	GUI.AIS.control_points,
+	GUI.AIS.control_points_remove,
+	GUI.AIS.correlation
+}
```

Comment:
Did you try doing table.unpack(GUI.AIS)?

Reply:
No. I don't know what that is. I'll have to look it up

Reply:
I'll look into the unpack function. However, things are working as-is, and this is already a big step forward on refactoring this, so let's not let table.unpack implementation hold up this PR. If I can get around to implementing it in the future I can do so.